core/src/client/mod.rs

```diff
@@ -399,13 +402,31 @@ pub enum CertificateStore {
 	WebPki,
 }
 
+/// JSON-RPC request object id data type.
+#[derive(Debug, Copy, Clone)]
+pub enum IdKind {
+	/// String.
+	String,
+	/// Number.
+	Number,
+}
+
+impl IdKind {
+	fn into_id(self, id: u64) -> Id<'static> {
+		match self {
+			IdKind::Number => Id::Number(id),
+			IdKind::String => Id::Str(format!("{}", id).into()),
+		}
+	}
+}
+
 #[cfg(test)]
 mod tests {
-	use super::RequestIdManager;
+	use super::{IdKind, RequestIdManager};
 
 	#[test]
 	fn request_id_guard_works() {
-		let manager = RequestIdManager::new(2);
+		let manager = RequestIdManager::new(2, IdKind::Number);
```

Comment:
Would it make sense to have a test where `IdKind::String`, not sure if this is covered elsewhere residually though.

Reply:
I think that's covered by some unit tests for the whole `ws client` but wouldn't hurt with a unit test for `String` also I suppose :)